tools/shared/src/main/scala/org/scalajs/core/tools/classpath/builder/JarTraverser.scala

```diff
@@ -61,9 +61,10 @@ trait JarTraverser extends ClasspathContentHandler with FileSystem {
     def entryBinaryContent: Array[Byte] = IO.readInputStreamToByteArray(in)
     def entryVersion: Option[String] = Some(entry.getTime.toString)
 
-    if (longName == JSDependencyManifest.ManifestFileName)
-      handleDepManifest(JSDependencyManifest.read(entryReader))
-    else if (longName.endsWith(".js")) {
+    if (longName == JSDependencyManifest.ManifestFileName) {
+      lazy val vf = JSDependencyManifest.read(entryReader)
+      handleDepManifest(vf)
```

Comment:
But it's true that the lazy behavior is not really needed. And a client who needs it can build it on its side.